sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -172,16 +176,15 @@ trait CheckAnalysis extends PredicateHelper {
             failAnalysis("Null-aware predicate sub-queries cannot be used in nested " +
               s"conditions: $condition")
 
+          case Filter(condition, _) if condition.find(isAggregateExpression(_)).isDefined =>
```

Comment:
call plan integrity checking here?

Reply:
@gatorsmile Thanks for reviewing. I will try it and get back.

Reply:
@gatorsmile @cloud-fan Just made change to make use of the integrity check method. I had to refactor it out as its a private method in optimizer. Is this what you had in mind ? Please let me know.